crates/topos-tce-broadcast/src/task_manager_channels/mod.rs

```diff
@@ -94,6 +94,12 @@ impl TaskManager {
 
                                     spawn(task.run());
 
+                                    if let Some(messages) = self.buffered_messages.remove(&cert.id) {
+                                        for msg in messages {
+                                            _ = task_context.sink.send(msg).await;
+                                        }
+                                    }
```

Comment:
```suggestion
                                    if let Some(messages) = self.buffered_messages.remove(&cert.id) {
                                    	let sink = task_context.sink.clone();
                                    	spawn(async move {
	                                    	for msg in messages {
	                                            _ = task_context.sink.send(msg).await;
	                                        }
                                    	});
                                    }
```

you can make it async too